packages/atomic/src/components/commerce/atomic-commerce-breadbox/atomic-commerce-breadbox.tsx

```diff
@@ -0,0 +1,435 @@
+        ],
+      }));
+  }
+
+  private get categoryFacetBreadcrumbs(): IBreadcrumb[] {
+    return (
+      this.breadcrumbManagerState.facetBreadcrumbs.filter(
+        ({facetId, type, values}) => {
+          return (
+            values.length > 0 &&
+            this.bindings.store.state.categoryFacets[facetId] &&
+            type === 'hierarchical'
+          );
+        }
+      ) as Breadcrumb<CategoryFacetValue>[]
+    ).map(({facetId, field, values}) => {
+      const deselectAll = () => {
+        values.forEach((value) => value.deselect());
+      };
+      return {
+        facetId,
+        label: this.bindings.store.state.categoryFacets[facetId].label(),
+        deselect: deselectAll,
+        formattedValue: values.map((pathValue) =>
+          getFieldValueCaption(field, pathValue.value.value, this.bindings.i18n)
+        ),
+      };
+    });
+  }
+
+  private get numericFacetBreadcrumbs(): IBreadcrumb[] {
+    return (
+      this.breadcrumbManagerState.facetBreadcrumbs.filter(
+        ({facetId, type}) =>
+          this.bindings.store.state.numericFacets[facetId] &&
+          type === 'numericalRange'
+      ) as Breadcrumb<NumericFacetValue>[]
+    )
+      .map(({facetId, field, values, type}) =>
+        values.map((value) => ({value, facetId, field, type}))
+      )
+      .flat()
+      .map(({value, facetId}) => ({
+        facetId,
+        label: this.bindings.store.state.numericFacets[facetId].label(),
+        state: value.value.state,
+        deselect: value.deselect,
+        formattedValue: [
+          this.bindings.store.state.numericFacets[facetId].format(value.value),
+        ],
+        content: this.bindings.store.state.numericFacets[facetId].content?.(
+          value.value
+        ),
+      }));
+  }
+
+  private get dateFacetBreadcrumbs(): IBreadcrumb[] {
+    return (
+      this.breadcrumbManagerState.facetBreadcrumbs.filter(
+        ({facetId, type}) =>
+          this.bindings.store.state.dateFacets[facetId] && type === 'dateRange'
+      ) as Breadcrumb<DateFacetValue>[]
+    )
+      .map(({facetId, field, values, type}) =>
+        values.map((value) => ({value, facetId, field, type}))
+      )
+      .flat()
+      .map(({value, facetId}) => ({
+        facetId,
+        label: this.bindings.store.state.dateFacets[facetId].label(),
+        state: value.value.state,
+        deselect: value.deselect,
+        formattedValue: [
+          this.bindings.store.state.dateFacets[facetId].format(
+            value.value as DateFacetValue
+          ),
+        ],
+      }));
+  }
+
+  private get allBreadcrumbs(): IBreadcrumb[] {
+    return [
+      ...this.facetBreadcrumbs,
+      ...this.categoryFacetBreadcrumbs,
+      ...this.numericFacetBreadcrumbs,
+      ...this.dateFacetBreadcrumbs,
+    ];
+  }
+
+  private renderBreadcrumbs(allBreadcrumbs: IBreadcrumb[]) {
+    const sortedBreadcrumbs = allBreadcrumbs.sort((a, b) => {
```

Comment:
Instead of building all breadcrumbs by type, and then sorting them, could we iterate over `this.breadcrumbManagerState.facetBreadcrumbs` and build them on the fly? I don't see the private `facetBreadcrumbs`, `categoryFacetBreadcrumbs`, `numericFacetBreadcrumbs` and `dateFacetBreadcrumbs` used elsewhere.

It's somewhat how we render breadcrumbs in barca: https://github.com/coveo/barca-sports/blob/672c05c55c02324d3db2381a5da8c1a32496dc8f/Components/Commerce/Breadcrumb.tsx#L42